### ISSUE TITLE
ImageColors: Illegal character in path 

### ISSUE THREAD

**Issue by bibhuti9** [osamaqarem/react-native-image-colors#66]
# Bug

## Environment info

| Library                   | Version |
| ------------------------- | ------- |
| react-native-image-colors | 1.5.2   |

when I try to get the image color for this path : "file:///storage/emulated/0/Images/Duck/Screenshot at Apr 29 08-52-35.png"

Error : ImageColors: Illegal character in path

**Comment by VigilioYonatan** [comment]
any solution?

**Comment by HubertRyanOfficial** [comment]
I've been having the same error, any solution until now?

**Comment by HubertRyanOfficial** [comment]
Hey guys, I was searching about how I could fix that and I got a solution to us. 🔥

Firstly I created a function to handle the photo URL and replace all spaces code in a string index 40. So the index 40 is a space code in a string. Look the function below:

`
export const URLEncoder = (path = '') => {
  if (isAndroid()) {
    return String(path).replace(/\s+/g, '%20');
  }

  return path;
};

`

So with this function you can import where the ImageColors is beeing used and put as parameter the photo url path, look it below: 

`
ImageColors.getColors(URLEncoder(uri), {
    fallback: Theme.colors.primary,
})
.then(res => {
    ...
})
`

In short, I'm using regex to remove all space to %20 in a string. Hope I've helped.

**Comment by osamaqarem** [comment]
> Hey guys, I was searching about how I could fix that and I got a solution to us. 🔥
> 
> Firstly I created a function to handle the photo URL and replace all spaces code in a string index 40. So the index 40 is a space code in a string. Look the function below:
> 
> ```
> export const URLEncoder = (path = '') => {
>   if (isAndroid()) {
>     return String(path).replace(/\s+/g, '%20');
>   }
> 
>   return path;
> };
> ```
> 
> So with this function you can import where the ImageColors is beeing used and put as parameter the photo url path, look it below:
> 
> `ImageColors.getColors(URLEncoder(uri), { fallback: Theme.colors.primary, }) .then(res => { ... })`
> 
> In short, I'm using regex to remove all space to %20 in a string. Hope I've helped.

I guess you should be able to use the standard `encodeURI(uri)` function instead of a custom one.
